resources/options.js

```diff
@@ -59,37 +67,17 @@ function restoreOptionsAvailability() {
         var enabled = items.loggingEnabled;
         if (enabled) {
             show();
+            restoreSubOptionsAvailability();
         } else {
             hide();
         }
     });
 }
 
-// Shows the sub-options.
-function showSubOptions() {
-    for (var i = 0; i < subOptions.length; i++) {
-        subOptionsElement(i).show();
-    }
-}
-
-// Hides the sub-options.
-function hideSubOptions() {
-    for (var i = 0; i < subOptions.length; i++) {
-        subOptionsElement(i).hide();
-    }
-}
-
-// Shows the option cards.
-function showCards() {
-    for (var i = 0; i < cards.length; i++) {
-        cardElement(i).show();
-    }
-}
-
-// Hides the option cards.
-function hideCards() {
-    for (var i = 0; i < cards.length; i++) {
-        cardElement(i).hide();
+// Restores the availability of the checkboxes, that are enabled / disabled by other checkboxes. using the logging value which is stored in chrome.storage.
+function restoreSubOptionsAvailability() {
+    for (var i = 0; i < onDisableHideAndDisable.length; i++) {
+        dynamicalyHideOrShowSubelements(onDisableHideAndDisable[i]);
```

Comment:
Please consistantly name your functions. Above you used SubOptions, here it is Subelements (single vs double capital)